app/src/main/java/io/apicurio/registry/storage/importing/v2/SqlDataUpgrader.java

```diff
@@ -97,7 +106,7 @@ public void importArtifactVersion(ArtifactVersionEntity entity) {
             io.apicurio.registry.utils.impexp.v3.ArtifactVersionEntity newEntity = io.apicurio.registry.utils.impexp.v3.ArtifactVersionEntity
                     .builder().createdOn(entity.createdOn).description(entity.description)
                     .labels(entity.labels != null
-                        ? entity.labels.stream().collect(Collectors.toMap(label -> label, label -> label))
+                        ? entity.labels.stream().collect(Collectors.toMap(label -> label, label -> null))
                         : Collections.emptyMap())
```

Comment:
For version labels I think we need to combine v2 labels (with `null` value as you are doing here) and v2 properties.  In the current code I think we're simply losing the properties data entirely.

It may be that we will still lose some data (in the case where a version has a label named `foo` and a property with name `foo` and value `bar`, resulting in a logical conflict).  But I think it's enough of an edge case that we shouldn't care.

In any case, the value of v3 `labels` should be a union of v2 `labels` and `properties`.

Note: same thing applies to creating the artifact below (line 123ish).